### PR TITLE
Test: refine astToExecutor to ExecutorBinders

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609

Problem Summary:

### What is changed and how it works?
Separate Executor structs from astToExecutor to Binders. Rename executors to binders.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Debug/MockExecutor/AstToPB.cpp [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+    /// check column
+    auto ft = checkSchema(input, id->getColumnName());
+    if (ft == input.end())
+        throw Exception("No such column: " + id->getColumnName(), ErrorCodes::NO_SUCH_COLUMN_IN_TABLE);
+    ci = ft->second;
+
+    return ci;
+}
+
+TiDB::ColumnInfo compilerFunction(const DAGSchema & input, ASTFunction * func)
```
Comment (xzhangxian1008) [inline dbms/src/Debug/MockExecutor/AstToPB.cpp:494]: may be compilerFunction -> compileFunction?

File: dbms/src/Debug/MockExecutor/AstToPB.cpp [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+        break;
+    }
+    for (const auto & child_ast : func->arguments->children)
+    {
+        compileExpr(input, child_ast);
+    }
+    return ci;
+}
+
+TiDB::ColumnInfo compilerLiteral(ASTLiteral * lit)
```
Comment (xzhangxian1008) [inline dbms/src/Debug/MockExecutor/AstToPB.cpp:618]: ditto

**Comment by xzhangxian1008** [comment]
lgtm

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**